src/Symfony/Bundle/FrameworkBundle/Routing/Attribute/AsRoutingConditionService.php

```diff
@@ -41,6 +41,10 @@
 #[\Attribute(\Attribute::TARGET_CLASS)]
 class AsRoutingConditionService extends AutoconfigureTag
 {
+    /**
+     * @param string|null $alias    The alias of the service to use it in routing condition expressions
+     * @param int         $priority The priority of the service when gathered in a service iterator or locator
```

Comment:
the "when" part is obscure to me, it tells about internal things, isn't it? what the use case for end users?

Reply:
Hi @renanbr, it seems you created this attribute. Could you please help me clearly understand what's the use of the `priority` parameter? I'm having a look at the code, here exactly:

https://github.com/symfony/symfony/pull/44405/files#diff-dad788051d1d21f8285f23cd25cbec3e9068393ae468fef0f359acd2112bea15R32

Because we're using `service(...)` to gather _one_ routing condition service at a time, I'm not sure to understand to goal of setting a priority on this for the end user?

Thanks!

Reply:
👋🏼 the goal is to allow overriding a service with the same alias

Reply:
Crystal clear. Thank you very much :pray: